BackendApartmentReservation/BackendApartmentReservation/Apartments/ApartmentRepository.cs

```diff
@@ -11,5 +16,44 @@ public ApartmentRepository(DatabaseContext db)
         {
             _db = db;
         }
+
+        public async Task<DbRoomReservation> CreateRoomReservation(DbEmployee employee, DateTime dateFrom, DateTime dateTo)
+        {
+            var roomReservation = new DbRoomReservation();
+            roomReservation.Employee = employee;
+            roomReservation.DateFrom = dateFrom;
+            roomReservation.DateTo = dateTo;
+
+            var availableRooms = new List<DbApartmentRoom>();
+            foreach (DbApartmentRoom room in _db.ApartmentRooms)
+            {
+                var isAvailable = _db.DbRoomReservations
+                    .Where(e => e.Room.Id == room.Id)
+                    .All(e => e.DateFrom.Date > dateTo.Date || e.DateTo.Date < dateFrom.Date);
```

Comment:
I think you should use `>=` and `<=` here, since there wont be any minutes or hours in the `DateTime`, only full days